### ISSUE TITLE
firmware T16

### ISSUE THREAD

**Issue by Javitol** [olivluca/zte-digi-admin#2]
Buenos días @olivluca , 

tu idea de simular el servidor tr es genial, muchas gracias por compartirla.

A mi me han puesto un ZTE H298A que se ha actualizado a la version T16 del firmware en cuanto el técnico lo enchufó.

En router conecta al puerto del servidor simulado, pero no envia el POST, igual ahora validan el certificado.

¿Alguna idea para conseguir el acceso admin con la version T16 del firmware?

Muchas gracias por todo.

**Comment by olivluca** [comment]
No, lo siento. Quizás si entras por puerto serie desde el bootloader puedes hacer downgrade a una versión anterior (que te puedo proporcionar) pero es posible que incluso el bootloader esté bloqueado.

**Comment by Javitol** [comment]
También lo intenté, en las 2 primeras opciones pide un "boot password" y en la tercera supongo que el root. La F601 tiene todos los puertos cerrados, supongo que en cuanto el técnico le configura el idont. Es curioso que dediquen tanto esfuerzo para que no puedas utilizar tus propios datos voip.

**Comment by olivluca** [comment]
Pues, no se me ocurre nada más. Quizás sea posible simular el router para que el tr069 provisione y así interceptar los datos sip, pero cuando hice mitm no conseguí que la central completase el aprovisionamiento, así que no tengo muchas pistas sobre como se pueda hacer.

**Comment by Javitol** [comment]
los datos que conseguiste de la central fueron por casualidad
ConnectionRequestPassword,ConnectionRequestUsername,PeriodicInformEnable,PeriodicInformInterval
?

Me da la sensación que primero configuran el router para poder conectarse con él y tras ello es cuando provisionan completamente con los datos sip, por eso no pudiste ver los datos interesantes con el mint.

**Comment by olivluca** [comment]
Exacto, a partir de ahí no provisionaba  si no quitaba el mitmproxy.
Simulando la misma secuencia no obtengo nada más del servidor.

**Comment by Javitol** [comment]
El router en el primer POST les envia su ConnectionRequestURL, su central guarda ese dato y le manda el ConnectionRequestUsername y el ConnectionRequestPassword, justo despues intentan desde la central conectar a esa ConnectionRequestURL con un GET, y si el router les devuelve un HTTP/1.1 200 OK, en ese momento un nuevo POST del router recibe como respuesta los datos SIP. 

Creo que la clave para ver los datos SIP en el mint es añadir una regla para acceder a la ConnectionRequestURL del router desde fuera.

**Comment by olivluca** [comment]
Pues, he enviado un bootstrap con una ConnectionRequestURL que controlo y no he visto ningún intento por su parte de conexión. Es posible que filtren e intenten conectar solo en un puerto que ellos esperan.

**Comment by olivluca** [comment]
Incluso si pongo la url del router no veo ningún intento (en el router tengo redirigidas las peticiones tr069 a mi servidor y si recibe un connection request lo veo).

**Comment by Javitol** [comment]
Yo lo vi creando un bridge y haciendo un reset de fabrica al router, en las trazas se veian las tramas HTTPS del primer POST del router, luego la conexión HTTP de la central contra el router para validar el cliente, y justo despues las tramas HTTPS del segundo POST del router

**Comment by olivluca** [comment]
Entonces de alguna manera la central detecta que no es el router y/o tiene alguna medida adicional de seguridad.

**Comment by olivluca** [comment]
por ejemplo es posible que la central solo haga eso si la sesión PPPoE se ha establecido con las credenciales genéricas digiuser@digi/123456

**Comment by Javitol** [comment]
Exacto, no envia los datos sip hasta que "valida" el router del cliente, y para validarlo intenta conectar contra él, configurandole el usuario y pass que es justo lo que veias con el mint.

**Comment by Javitol** [comment]
Esas credenciales genéricas se usan para acceder al servidor TR, si te fijas al acceder con ellas solo hay acceso a su red local de servidores, no dan acceso externo a internet.

**Comment by olivluca** [comment]
Ya, por eso digo, probablemente si tienes una sessión PPPoE _real_ la central ni siquiera intenta aprovisionar, solo lo hace si usas las credenciales genéricas (probablemente tenga un rango de ip especifico).

1. el router envía el mensaje de bootstrap
2. el servidor envía los parámetros para el connection request
3. **solo si la conexión procede de la sesión PPPoE genérica** el servidor hace la GET al router para provocar un connection request.
4. el router ejecuta el connecion request y recibe la configuración completa

Si pudiese quedarme sin internet lo intentaría (con un cliente PPPoE en mi ordenador, no pienso hacer un reset al router :wink:)

**Comment by Javitol** [comment]
Exacto, funciona tal y como indicas :)
Suerte que no te han puesto la T16 o tienes un boot que no te pide clave para hacer downgrade.

Habia pensado tambien en la descarga de firmwares 
http://tr069-1.s.digimobil.es:9091/firmwareFiles/
va sin ssl aunque pide clave, aunque no creo que sea dificil de conseguir y se podría redireccionar a un server simulado con firmwares antiguos.

**Comment by olivluca** [comment]
De ahí he sacado todos los firmwares desde la T2 hasta la T16 :wink:

**Comment by dw1247889** [comment]
Antes de todo, muchas gracias por todo el trabajo .

No sé si se ha conseguido alguna cosa con el firmware T16. Yo he probado cosas como Javitol pero sin éxito.

¿Sabéis si hay algún dump del firmware de digi? Con binwalk no he visto la manera de poderlo conseguir.

Un saludo y muchas gracias,

**Comment by olivluca** [comment]
Por mi parte no hay ninguna novedad

**Comment by UltrashRicco** [comment]
Hola, pues parece que publicaron el FW T17 a principios de año.
Tengo el router Digi sustituido por un DLINK, y el router Digi tan sólo me sirve para la telefonía VOIP.
He reiniciado el router Digi a configuración de fábrica, lo he conectado con la ONT para dejarlo actualizarse a la T17.
He podido volver a acceder como admin con la contraseña admin de fábrica, que se encuentra dentro del config.bin. Si se cambió la contraseña admin, la contraseña admin de fábrica es la misma que la del usuario root.
No he notado cambios dentro de las opciones de configuración disponibles, no sé qué cambios introduce el T17
![2022-03-19_12h43_33](https://user-images.githubusercontent.com/54411674/159122206-5c001979-5d07-47dc-a983-eba9fd50de30.png)
![2022-03-19_13h15_07](https://user-images.githubusercontent.com/54411674/159122230-e26afd88-45b9-4d31-8050-c980be42326b.png)
.

**Comment by kokoparkletal** [comment]
Buenas, estoy probando con la T17 y no consigo nada...

He probado a cargar el T11 y T14 ruso que he encontrado para hacer downgrade y me da error como era de esperar.

Algún alma caritativa tiene un T12 o anterior de digi a ver si me pasa el checksum?

Por otro lado la contraseña por defecto cuántos caracteres tiene? Tiene símbolos? Es solo hexadecimal o alfanumérica? Que lo mismo me da por probar con fuerza bruta a ver qué pasa, y si mi quito caracteres mejor.

**Comment by olivluca** [comment]
Si no has podido hacer el downgrade con la T11 dudo que puedas con la T12, pero por tener la tengo.

**Comment by kokoparkletal** [comment]
Al final lo conseguí con la T2 de digi, el problema es que la versión rusa que encontré daba error de checksum, aparentemente a partir de alguna versión solo se puede downgradear a versiones de digi.
Muchas gracias.

**Comment by fagmixy** [comment]
@olivluca 
Hello i want admin password for digi zte h3600 any idea ????

**Comment by olivluca** [comment]
> @olivluca Hello i want admin password for digi zte h3600 any idea ????

no, sorry.

**Comment by asterlxx** [comment]
Hola
buenas
es posible hacer algo similar para el  ZXHN H298Q ?
gracias

**Comment by olivluca** [comment]
Lo dudo, DIGI se esfuerza mucho para que no puedas sacar los datos del VOIP, Si han parcheado el H298A para validar el certificado del servidor, lo más probable es que el H298Q también lo haga. Es cuestión de probar.
Ahora, actualizaciones para fallos de seguridad y/o de funcionamiento tardan bastante más en llegar (si es que llegan) 😡 .

**Comment by asterlxx** [comment]
en verdad me interesa tener el usuario admin.. solo eso.. porque consegui un router de digi.. pero no tengo digi :P
y queria ver si se podia poner un firmware defaul del modelo sin marca..
sabes algo?

**Comment by olivluca** [comment]
no, lo siento
